### PR TITLE
Update protos and attribute mapping

### PR DESCRIPTION
**Pull request by bryannaegele**
The latest protos introduced a breaking change for how attributes are defined. They also added attribute value support for arrays and maps. This fixes handling of attribute mapping and adds base support for proplists, maps, and lists for the new value types.

### CODE REVIEW

File: rebar.lock [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{"1.2.0",
```
Comment (tolbrino) [inline rebar.lock:1]: So you used an older version of rebar here?

File: src/opentelemetry_exporter.erl [hunk 2]
```diff
@@ -145,27 +145,40 @@ to_unixnano(Timestamp) ->
...
+        false ->
+            to_attributes(Rest, [#{key => Key,
+                                   values => #{value => {array_value, Value}}} | Acc])
+    end;
 to_attributes([_ | Rest], Acc) ->
     to_attributes(Rest, Acc).
 
+is_proplist([]) -> true;
+is_proplist([{K,_}|L]) when is_atom(K) or is_binary(K) -> is_proplist(L);
+is_proplist(_) -> false.
```
Comment (tolbrino) [inline src/opentelemetry_exporter.erl:180]: Should lists with values of different types be discarded as well? The code above looks like its meant to handle proplists and lists with simple types only.

File: rebar.lock (hunk 1 above)
Comment (bryannaegele) [reply rebar.lock:1]: I have the latest release `3.13.2`. I think @tsloughter last ran it with an rc version of `3.14.0`.

https://github.com/erlang/rebar3/releases

File: src/opentelemetry_exporter.erl (hunk 2 above)
Comment (bryannaegele) [reply src/opentelemetry_exporter.erl:180]: The key has to be a binary or atom. The values can be somewhat arbitrary (see the note on nested maps), but they should be simple types in practice. The initial goal was just distinguishing that the list was a true proplist and not a mixed list, which has to be dropped.

File: rebar.lock (hunk 1 above)
Comment (tsloughter) [reply rebar.lock:1]: If you didn't do any dep upgrades/changes can you reset this file to the original.

**Comment by tsloughter** [comment]
I'll approve and merge once the lock file is reset.

**Pull request merged**